beacon-chain/slasher/queue.go

```diff
@@ -0,0 +1,83 @@
+package slasher
+
+import (
+	"sync"
+
+	slashertypes "github.com/prysmaticlabs/prysm/beacon-chain/slasher/types"
+)
+
+// Struct for handling a thread-safe list of indexed attestation wrappers.
+type attestationsQueue struct {
+	lock  sync.RWMutex
```

Comment:
You can just do `sync.RWMutex`

Then you can just do `q.Lock()` or `q.UnLock()`